docs/how-to-guides/executable-with-internal-dependencies.rst

```diff
@@ -0,0 +1,20 @@
+.. _how-to-guides-executable-with-internal-dependencies:
+
+*************************************
+Executable with internal dependencies
+*************************************
+
+
+.. todo::
+
```

Comment:
The macOS PR for this is about ready to merge, and we can probably sort Windows out soon. So maybe the static library explanation isn't needed?

Reply:
Well, it's already written :rofl:. We can just keep it and remove when it's fixed, I don't think it'd be much trouble.